src/mini-batch.jl

```diff
@@ -0,0 +1,73 @@
+"""
+    MiniBatch(b::Int)
+
+"""
+struct MiniBatch <: AbstractKMeansAlg
+    b::Int  # batch size
+end
+
+
+MiniBatch() = MiniBatch(100)
+
+function km(algo::MiniBatch, X, k;
+                weights = nothing, metric = Euclidean(), n_threads = Threads.nthreads(),
+                k_init = "k-means++", init = nothing, max_iters = 300,
+                tol = 0, verbose = false, rng = Random.GLOBAL_RNG)
+
+    # Step 1. Select sample from X as specified by batch_size and weights
+    nrow, ncol = size(X)  # n_features, m_examples
+    T = eltype(X)
+    rand_idx = isnothing(weights) ? rand(rng, 1:ncol, algo.b) : wsample(rng, 1:ncol, weights, algo.b)
+    initial_batch_sample = X[:, rand_idx]
```

Comment:
No need to allocate a new sample. You can just use $rand_idx$ everywhere.
Theoretically, it is possible that allocated data works faster, but it should be verified (and I do not think so, since we are sampling columns)